scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -259,6 +267,46 @@ public void run() {
         return true;
     }
 
+    /**
+     * Create a new Credential object containing users' 3rd Party Credentials.
+     *
+     * @param creds credentials for specific user
+     * @return in case of success new object containing the 3rd party credentials used to create bindings
+     * at clean script
+     */
+    Credentials addThirdPartyCredentials(Credentials creds) throws KeyException, IllegalAccessException {
+        //retrieve scheduler key pair
+        String privateKeyPath = PASchedulerProperties.getAbsolutePath(PASchedulerProperties.SCHEDULER_AUTH_PRIVKEY_PATH.getValueAsString());
```

Comment:
create a private method like this : 
PrivateKey privateKey = getKey(PASchedulerProperties.SCHEDULER_AUTH_PRIVKEY_PATH.getValueAsString())
PrivateKey privateKey = getKey(PASchedulerProperties. SCHEDULER_AUTH_PUBKEY_PATH.getValueAsString())

Reply:
Ok but it will be called only once! I cannot use the same for public and private key because the getter is different.